### ISSUE TITLE
Cookbook does not support CentOS 7

### ISSUE THREAD

**Issue by hhoover** [edelight/chef-mongodb#330]
Using the default recipe/test suite:

```
================================================================================
Error executing action `start` on resource 'service[mongod]'
================================================================================


       Errno::ENOENT

       -------------
       No such file or directory - /sbin/service mongod start


       Resource Declaration:
       ---------------------
       # In /tmp/kitchen/cookbooks/mongodb/definitions/mongodb.rb

       191:   service new_resource.name do
       192:     provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
       193:     supports :status => true, :restart => true
       194:     action new_resource.service_action
       195:     new_resource.service_notifies.each do |service_notify|
       196:       notifies :run, service_notify
       197:     end
       198:     notifies :create, 'ruby_block[config_replicaset]' if new_resource.is_replicaset && new_resource.auto_configure_replicaset
       199:     notifies :create, 'ruby_block[config_sharding]', :immediately if new_resource.is_mongos && new_resource.auto_configure_sharding
       200:       # we don't care about a running mongodb service in these cases, all we need is stopping it
       201:     ignore_failure true if new_resource.name == 'mongodb'
       202:   end
       203:
       204:   # replicaset
       205:   if new_resource.is_replicaset && new_resource.auto_configure_replicaset
       206:     rs_nodes = search(
       207:       :node,
       208:       "mongodb_cluster_name:#{new_resource.replicaset['mongodb']['cluster_name']} AND \
       209:        mongodb_is_replicaset:true AND \
       210:        mongodb_shard_name:#{new_resource.replicaset['mongodb']['shard_name']} AND \
       211:        chef_environment:#{new_resource.replicaset.chef_environment}"
       212:     )
       213:
       214:     ruby_block 'config_replicaset' do
       215:       block do
       216:         MongoDB.configure_replicaset(new_resource.replicaset, replicaset_name, rs_nodes) unless new_resource.replicaset.nil?
       217:       end
       218:       action :nothing
       219:     end
       220:
       221:     ruby_block 'run_config_replicaset' do
       222:       block {}
       223:       notifies :create, 'ruby_block[config_replicaset]'
       224:     end
       225:   end
       226:



       Compiled Resource:
       ------------------
       # Declared in /tmp/kitchen/cookbooks/mongodb/definitions/mongodb.rb:191:in `block in from_file'

       service("mongod") do
         params {:mongodb_type=>"mongod", :action=>[:enable, :start], :logpath=>"/var/log/mongodb/mongodb.log", :dbpath=>"/var/lib/mongodb", :configservers=>[], :replicaset=>nil, :notifies=>[], :bind_ip=>"0.0.0.0", :port=>27017, :enable_rest=>false, :smallfiles=>false, :name=>"mongod"}
         action [:enable, :start]
         supports {:status=>true, :restart=>true}
         retries 0
         retry_delay 2
         service_name "mongod"
         enabled true

         pattern "mongod"
         startup_type :automatic
         cookbook_name :mongodb
         recipe_name "default"
       end
```

**Comment by jamesonjlee** [comment]
awww, will look into it.

**Comment by GUI** [comment]
I've submitted a pull request which I believe fixes things under CentOS 7: #350
